runtime/drivers/admin.go

```diff
@@ -3,14 +3,21 @@ package drivers
 import (
 	"context"
 	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
 )
 
 type AdminService interface {
-	GetReportMetadata(ctx context.Context, reportName string, annotations map[string]string, executionTime time.Time) (*ReportMetadata, error)
+	GetReportMetadata(ctx context.Context, reportName string, reportSpec *runtimev1.ReportSpec, executionTime time.Time) (*ReportMetadata, error)
 	GetAlertMetadata(ctx context.Context, alertName string, annotations map[string]string, queryForUserID, queryForUserEmail string) (*AlertMetadata, error)
 }
 
 type ReportMetadata struct {
+	InternalUsersURL ReportURLs
+	ExternalUsersURL map[string]ReportURLs
```

Comment:
nit: what about naming the fields like this:

```suggestion
	BaseURLs ReportURLs
	RecipientURLs map[string]ReportURLs
```

Since the base/fallback URLs are also used for things like Slack destinations, which are not really a user recipient